pages/id/[form]/users.tsx

```diff
@@ -0,0 +1,144 @@
+        formID,
+        users,
+      },
+      timeout: process.env.NODE_ENV === "production" ? 60000 : 0,
+    });
+  } catch (e) {
+    logMessage.error(e);
+  }
+};
+
+const usersToOptions = (users: User[]): { value: string; label: string | null }[] => {
+  return users.map((user) => {
+    return { value: user.id, label: user.name };
+  });
+};
+
+const Users = ({
+  formRecord,
+  usersAssignedToFormRecord,
+  allUsers,
+}: AssignUsersToTemplateProps): React.ReactElement => {
+  const { t, i18n } = useTranslation("admin-users");
+  const language = i18n.language as string;
+  const router = useRouter();
+
+  const [selectedUsers, setSelectedUsers] = useState<{ value: string; label: string | null }[]>(
+    usersToOptions(usersAssignedToFormRecord)
+  );
+
+  const saveAssignations = async () => {
+    const usersToAdd: { id: string; action: "add" | "remove" }[] = selectedUsers
+      .filter((user) => {
+        return !usersAssignedToFormRecord.map((u) => u.id).includes(user.value);
+      })
+      .map((user) => {
+        return { id: user.value, action: "add" };
+      });
+
+    const usersToRemove: { id: string; action: "add" | "remove" }[] = usersAssignedToFormRecord
+      .filter((user) => {
+        return !selectedUsers.map((u) => u.value).includes(user.id);
+      })
+      .map((user) => {
+        return { id: user.id, action: "remove" };
+      });
+
+    await updateUsersToTemplateAssignations(formRecord.id, usersToAdd.concat(usersToRemove));
+    await router.push({ pathname: "/admin/view-templates" });
+  };
+
+  return (
+    <>
+      <h1 className="gc-h1">{formRecord.form[getProperty("title", language)] as string}</h1>
+      <div className="mb-4">{t("assignUsersToTemplate")}</div>
+      <Select
+        isClearable
+        isSearchable
+        isMulti
+        components={makeAnimated()}
+        options={usersToOptions(allUsers)}
+        value={selectedUsers}
+        onChange={(value) => setSelectedUsers(value as { value: string; label: string | null }[])}
+      />
+      <br />
+      <Button className="" type="submit" onClick={() => saveAssignations()}>
+        {t("save")}
+      </Button>
+    </>
+  );
+};
+
+const redirect = (locale: string | undefined) => {
+  return {
+    redirect: {
+      // We can redirect to a 'Form does not exist page' in the future
+      destination: `/${locale}/404`,
+      permanent: false,
+    },
+  };
+};
+
+export const getServerSideProps = requireAuthentication(
+  async ({ user: { ability }, locale, params }) => {
+    checkPrivileges(ability, [
+      { action: "update", subject: "FormRecord" },
+      { action: "update", subject: "User" },
+    ]);
+
+    const formID = params?.form;
+    if (!formID || Array.isArray(formID)) return redirect(locale);
+
+    const templateWithAssociatedUsers = await getTemplateWithAssociatedUsers(ability, formID);
+    if (!templateWithAssociatedUsers) return redirect(locale);
+
+    const allUsers = await getUsers(ability);
```

Comment:
I'd be careful about passing `allUsers` directly to the page props as it contains much more information about a user than a the page needs.
I'd suggest parsing out the properties needed by the page `{id, name}` before passing the object down as props.

Reply:
Did change it :)